massa-execution/src/interface_impl.rs

```diff
@@ -29,15 +38,73 @@ impl Interface for InterfaceImpl {
     fn get_module(&self, address: &String) -> Result<Vec<u8>> {
         let bytecode = {
             let address = Address::from_str(address)?;
-            let context = self
-                .context
-                .lock()
-                .expect("Failed to acquire lock on context.");
+            let context = context_guard!(self);
             context.ledger_step.get_module(&address)
         };
         match bytecode {
             Some(bytecode) => Ok(bytecode),
             _ => bail!("Error bytecode not found"),
         }
     }
+
+    /// Requires a new address that contains the sent bytecode.
+    ///
+    /// Generate a new address with a concatenation of the firsts N bytes of the blockId and the firsts N bytes of the block creator address.
+    /// Where N = ADDRESS_SIZE_BYTES / 2
+    ///
+    /// Insert in the ledger the given bytecode in the generated address
+    fn create_module(&self, module: &Bytecode) -> Result<assembly_simulator::Address> {
+        let (block_id, index) = {
+            let context = context_guard!(self);
+            (context.opt_block_id, context.call_stack.len())
```

Comment:
index should not be the length of the call stack, but the index of the ledger entry creation within the block

Reply:
You can chck the new version. The next index is the number of owned addresses.
Implemented like in #2051

Reply:
good idea !